sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSenderTest.java

```diff
@@ -117,6 +130,76 @@ public void testLinkSize() throws IOException {
         verify(sender, times(1)).getRemoteMaxMessageSize();
     }
 
+    @Test
+    public void testCompleteTransaction() {
+        final String txnId = "1";
+        Declared transactionState = new Declared();
+        transactionState.setTxnId(Binary.create(ByteBuffer.wrap(txnId.getBytes())));
+
+        AmqpTransaction transaction = new AmqpTransaction(ByteBuffer.wrap("1".getBytes()));
+
+        ReactorSender reactorSender = new ReactorSender(entityPath, sender, handler, reactorProvider, tokenManager,
+            messageSerializer, Duration.ofSeconds(1), new ExponentialAmqpRetryPolicy(new AmqpRetryOptions()));
+        ReactorSender spyReactorSender = spy(reactorSender);
+
+        doReturn(Mono.just(transactionState)).when(spyReactorSender).send(any(byte[].class), anyInt(), anyInt(), isNull());
+
+        spyReactorSender.completeTransaction(transaction, true).subscribe();
```

Comment:
`subscribe()` is non-blocking. It may not have finished by the time verify line is hit.

Reply:
These tests are not using the right way to test Project Reactor.

https://projectreactor.io/docs/core/release/reference/#_testing_a_scenario_with_stepverifier